### ISSUE TITLE
Fix documentation of matchaddpos()

### ISSUE THREAD

**Issue by cwfoo** [vim/vim#7907]
The documentation of [`matchaddpos()`](https://github.com/vim/vim/blob/9faec4e3d439968e21ad74e917aebb289df8f849/runtime/doc/eval.txt#L7444-L7479) is confusing:

1. The documentation says that the function is:
    ```
    matchaddpos({group}, {pos} [, {priority} [, {id} [, {dict}]]])
    ```
    A better description would be:
    ```
    matchaddpos({group}, {list} [, {priority} [, {id} [, {dict}]]])
    ```
    where `{list}` is a list of positions.
2. The description of `{pos}` is wrong. The documentation says: "The list {pos} can contain one of these items: A number ... A list with one number ... A list with two numbers ...". Unfortunately, this is contradicted by the example given in the documentation. It is wrong because the list can contain **zero to eight** items (not just one).
3. Missing word somewhere here:
https://github.com/vim/vim/blob/9faec4e3d439968e21ad74e917aebb289df8f849/runtime/doc/eval.txt#L7474-L7476

    There's probably a missing word somewhere in this: " ... with the value a list like the {pos} item."

---

Proposed fix:

```diff
diff --git a/runtime/doc/eval.txt b/runtime/doc/eval.txt
index 03fcb655d..574edf707 100644
--- a/runtime/doc/eval.txt
+++ b/runtime/doc/eval.txt
@@ -2708,7 +2708,7 @@ match({expr}, {pat} [, {start} [, {count}]])
                                Number  position where {pat} matches in {expr}
 matchadd({group}, {pattern} [, {priority} [, {id} [, {dict}]]])
                                Number  highlight {pattern} with {group}
-matchaddpos({group}, {pos} [, {priority} [, {id} [, {dict}]]])
+matchaddpos({group}, {list} [, {priority} [, {id} [, {dict}]]])
                                Number  highlight positions with {group}
 matcharg({nr})                 List    arguments of |:match|
 matchdelete({id} [, {win}])    Number  delete match identified by {id}
@@ -7442,15 +7442,15 @@ matchadd({group}, {pattern} [, {priority} [, {id} [, {dict}]]])
                        GetGroup()->matchadd('TODO')
 <
                                                        *matchaddpos()*
-matchaddpos({group}, {pos} [, {priority} [, {id} [, {dict}]]])
-               Same as |matchadd()|, but requires a list of positions {pos}
+matchaddpos({group}, {list} [, {priority} [, {id} [, {dict}]]])
+               Same as |matchadd()|, but requires a list of positions
                instead of a pattern. This command is faster than |matchadd()|
                because it does not require to handle regular expressions and
                sets buffer line boundaries to redraw screen. It is supposed
                to be used when fast match additions and deletions are
                required, for example to highlight matching parentheses.
 
-               The list {pos} can contain one of these items:
+               {list} should be a list of positions, where a position is:
                - A number.  This whole line will be highlighted.  The first
                  line has number 1.
                - A list with one number, e.g., [23]. The whole line with this
@@ -7463,7 +7463,7 @@ matchaddpos({group}, {pos} [, {priority} [, {id} [, {dict}]]])
                - A list with three numbers, e.g., [23, 11, 3]. As above, but
                  the third number gives the length of the highlight in bytes.
 
-               The maximum number of positions is 8.
+               The maximum number of positions in {list} is 8.
 
                Example: >
                        :highlight MyGroup ctermbg=green guibg=green
@@ -7473,7 +7473,7 @@ matchaddpos({group}, {pos} [, {priority} [, {id} [, {dict}]]])
 
 <              Matches added by |matchaddpos()| are returned by
                |getmatches()| with an entry "pos1", "pos2", etc., with the
-               value a list like the {pos} item.
+               value being a position (like an item in {list}).
 
                Can also be used as a |method|: >
                        GetGroup()->matchaddpos([23, 11])
```

Vim version: 8.1.2269

**Comment by andymass** [comment]
The description is unclear but it's not completely wrong.  The list {pos} is a list of up to 8 items where *each item* is either:
- a number
- a list of one number
- a list of two numbers
- a list of three number

**Comment by brammool** [comment]
Thanks, I'll include the changes and tweak it a bit more.

**Comment by cwfoo** [comment]
Fixed in c8cdf0f80b3cfd88a4490d80572062c1fd1a96ca
